packages/@aws-cdk/aws-redshift/lib/database-query-provider/create-table.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable-next-line import/no-unresolved */
+import * as AWSLambda from 'aws-lambda';
+import { ClusterProps, executeStatement, getClusterPropsFromEvent, getResourceNameFromPhysicalId, getResourceProperty, makePhysicalId } from './util';
+
+interface Column {
+  name: string;
+  dataType: string;
+}
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const tableColumns = JSON.parse(getResourceProperty('tableColumns', event.ResourceProperties)) as Column[];
+  const tableName = event.ResourceProperties.tableName;
+  const clusterProps = getClusterPropsFromEvent(event.ResourceProperties);
+
+  if (event.RequestType === 'Create') {
+    await createTable(tableName, tableColumns, clusterProps);
+    return { PhysicalResourceId: makePhysicalId(tableName, clusterProps), Data: { tableName: tableName } };
+  } else if (event.RequestType === 'Delete') {
+    await dropTable(getResourceNameFromPhysicalId(event.PhysicalResourceId), clusterProps);
+    return;
+  } else if (event.RequestType === 'Update') {
+    await updateTable(tableName, tableColumns, clusterProps, event.OldResourceProperties);
+    return { PhysicalResourceId: makePhysicalId(tableName, clusterProps), Data: { tableName: tableName } };
```

Comment:
🚨 Be aware that if you return a different physicalId, CloudFormation will send you a `Delete` event for the table during the `STACK_CLEANUP` phase.